state-sync/storage-service/server/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+            },
+            data_summary: self.storage.get_data_summary()?,
+        };
+
+        Ok(StorageServiceResponse::StorageServerSummary(
+            storage_server_summary,
+        ))
+    }
+
+    fn get_transactions_with_proof(
+        &self,
+        request: TransactionsWithProofRequest,
+    ) -> Result<StorageServiceResponse, Error> {
+        let transactions_with_proof = self.storage.get_transactions_with_proof(
+            request.proof_version,
+            request.start_version,
+            request.expected_num_transactions,
+            request.include_events,
+        )?;
+
+        Ok(StorageServiceResponse::TransactionsWithProof(
+            transactions_with_proof,
+        ))
+    }
+}
+
+/// The interface into local storage (e.g., the Diem DB) used by the storage
+/// server to handle client requests.
+pub trait StorageReaderInterface {
+    /// Returns a data summary of the underlying storage state.
+    fn get_data_summary(&self) -> Result<DataSummary, Error>;
+
+    /// Returns a list of transactions with a proof relative to the
+    /// `proof_version`. The transaction list is expected to contain *at most*
+    /// `expected_num_transactions` and start at `start_version`.
+    /// If `include_events` is true, events are also returned.
+    fn get_transactions_with_proof(
+        &self,
+        proof_version: u64,
+        start_version: u64,
+        expected_num_transactions: u64,
+        include_events: bool,
+    ) -> Result<TransactionListWithProof, Error>;
+
+    /// Returns a list of epoch ending ledger infos, starting at `start_epoch`
+    /// and ending *at most* at the `expected_end_epoch`.
+    fn get_epoch_ending_ledger_infos(
+        &self,
+        start_epoch: u64,
+        expected_end_epoch: u64,
+    ) -> Result<EpochChangeProof, Error>;
+
+    // TODO(joshlind): support me!
+    //
+    // Returns a list of transaction outputs with a proof relative to the
+    // `proof_version`. The transaction output list is expected to contain
+    // *at most* `expected_num_transaction_outputs` and start at `start_version`.
+    //fn get_transaction_outputs_with_proof(
+    //    &self,
+    //    proof_version: u64,
+    //    start_version: u64,
+    //    expected_num_transaction_outputs: u64,
+    //) -> Result<TransactionOutputListWithProof, Error>;
+
+    // TODO(joshlind): support me!
+    //
+    // Returns an AccountStateChunk holding a list of account states
+    // starting at the specified account key with *at most*
+    // `expected_num_account_states`.
+    //fn get_account_states_chunk(
+    //    version,
+    //    start_account_key,
+    //    expected_num_account_states: u64,
+    //) -> Result<AccountStateChunk, Error>
+
+    // TODO(joshlind): identify the exact API here to get the frozen subtrees!
+    //
+    // Returns the frozen subtrees at the specified version.
+    //fn get_frozen_subtrees(
+    //    version,
+    //) -> Result<Vec<HashValue>, Error>
```

Comment:
What is this API for? bootstrapping the `TransactionAccumulatorSummary`?

Reply:
It's for the case where a node grabs all account states and needs to bootstrap it's view of the transaction accumulator. However, I'll remove it for now until I'm absolutely sure we need it. I think there's a way around it.